Solving default frame consistency in config files

AlexisTM opened the pull request:
Solving default frame consistency in config files #789 

* Changing the `local_origin` frame to `map` for consistency.
* Correction of the `global_position/frame_id` and `local_position/frame_id` to map frame

AlexisTM commented:
I updated the APM config as well. @TSC21

TSC21 commented:
Where you see `fcu_utm` please update to `fcu`

AlexisTM commented:
Should check for the time reference. 

Is this related to tf?

```
time:
  time_ref_source: "fcu"  # time_reference source
```

TSC21 commented:
> Should check for the time reference. 

Keep it as `fcu`. That has nothing to do with frames of reference.

TSC21 commented on mavros/launch/px4_config.yaml:
```diff
@@ -44,18 +44,18 @@ cmd:
     send: false               # send TF?
-    frame_id: "local_origin"  # TF frame_id
+    frame_id: "map"  # TF frame_id
     global_frame_id: "earth"  # TF earth frame_id
```
Can you add `global_frame_id` also to apm_config please?

TSC21 commented:
@AlexisTM looking good but can you please run a sanity check on the source code also and match the variables, that represent the parameters, to the config files? Basically the same you did above but for the cpp files. Thanks

AlexisTM commented:
The only parameters not synced are in `sys_status.cpp`

https://github.com/mavlink/mavros/blob/d283fb0a3f094caa4282922351fb965f69f9c898/mavros/src/plugins/sys_status.cpp#L431-L433

* `global_position/tf/send` default to `false`
* `imu, checked
* `local_position/tf/send` default to `false`
* `local_position/tf/send_fcu` default to `false`
* `mission/pull_after_gcs` default to `true`

TSC21 commented:
> The only parameters not synced are in `sys_status.cpp`

Can you sync with the values that are on the `.cpp` file, please?

> Note that the builds are failing because of the MAVLink signing development.

Yeah we need that the release comes out on the deb pakages.

AlexisTM commented:
* Added sys_status, and mavros_extras plugins.
* Added the FOV for the Px4flow (6.8 degrees at 5 meters)
* Adding the copter_visualization configuration in the yaml file

TSC21 commented:
@AlexisTM `copter_visualization` is a separate node that can be run in parallel. So the parameters are  not loaded from the config files of the `mavros` node. The parameter config should got: or to a `launch` file that launches the `copter_visualization` node, or to a `yaml` file on the launch folder and getting the parameters on the cpp file through `XmlRpc::XmlRpcValue` (check how it is done on `servo_state_publisher.cpp`).
The rest of the changes though seem consistent. Thanks

TSC21 commented:
Also, can you please change the `vibration` `frame_id` on both plugin and yaml to `base_link`? Thanks

AlexisTM commented:
I think it is all fine with the way the parameters are read for the `copter_visualization`. Indeed, there are only a finite number of parameters, I would propose to avoid the use of `XmlRpc::XmlRpcValue`. 

Setting the parameters in a yaml file or the launch file is sufficient.

TSC21 commented on mavros/launch/apm_config.yaml:
```diff
@@ -167,7 +169,7 @@ fake_gps:
 px4flow:
   frame_id: "px4flow"
-  ranger_fov: !degrees 0.0  # XXX TODO
+  ranger_fov: !degrees 6.8  # XXX TODO
```
You can remove the `TODO` comment now

TSC21 replied:
`6.8 degreens at 5 meters, 31 degrees at 1 meter`

TSC21 commented on mavros_extras/launch/px4flow_config.yaml:
```diff
@@ -25,7 +25,7 @@ image:
 px4flow:
   frame_id: "px4flow"
-  ranger_fov: !degrees 0.0  # XXX TODO
+  ranger_fov: !degrees 6.8  # XXX TODO
```
`6.8 degreens at 5 meters, 31 degrees at 1 meter`

TSC21 commented on mavros_extras/src/plugins/px4flow.cpp:
```diff
@@ -45,7 +45,9 @@ class PX4FlowPlugin : public plugin::PluginBase {
 
 		/** Default rangefinder is Maxbotix HRLV-EZ4 */
-		flow_nh.param("ranger_fov", ranger_fov, 0.0);	/** @todo Check MAxbotix HRLV-EZ4 Field-of-View */
+		flow_nh.param("ranger_fov", ranger_fov, 0.119428926);	/** @todo Check MAxbotix HRLV-EZ4 Field-of-View 
```
Replace with
```c++
/**
 * @note This is a narrow beam (60cm wide at 5 meters,
 * but also at 1 meter) 6.8 degrees at 5 meters, 31 degrees at 1 meter
 */
flow_nh.param("ranger_fov", ranger_fov, 0.119428926);
```

AlexisTM commented:
You are welcome. ;)
